nucypher/policy/reservoir.py

```diff
@@ -0,0 +1,115 @@
+"""
+This file is part of nucypher.
+
+nucypher is free software: you can redistribute it and/or modify
+it under the terms of the GNU Affero General Public License as published by
+the Free Software Foundation, either version 3 of the License, or
+(at your option) any later version.
+
+nucypher is distributed in the hope that it will be useful,
+but WITHOUT ANY WARRANTY; without even the implied warranty of
+MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+GNU Affero General Public License for more details.
+
+You should have received a copy of the GNU Affero General Public License
+along with nucypher.  If not, see <https://www.gnu.org/licenses/>.
+"""
+from typing import Iterable, Optional, List
+
+from eth_typing import ChecksumAddress
+
+from nucypher.blockchain.eth.agents import StakersReservoir, StakingEscrowAgent
+
+
+def make_federated_staker_reservoir(learner: 'Learner',
```

Comment:
May be enough to just pass the `FleetSensor` object here

Reply:
✔️